### PR TITLE
Fix for #2322 - ensure impact layer is added just above exposure layer

### PR DESCRIPTION
**Pull request by timlinux**
Note I expect a couple of failed tests - will fix that in a follow up PR. In the mean time, it would be great if @adissadis and @samnawi could try this out and see if it works ok for you.

### CODE REVIEW

**Comment by samnawi** [comment]
@timlinux LGTM
+1 for this

File: safe/gui/widgets/dock.py [hunk 1]
```diff
@@ -1319,6 +1322,77 @@ def prepare_analysis(self):
 
         return analysis
 
+    def add_above_layer(
```
Comment (ismailsunni) [inline safe/gui/widgets/dock.py]: I think it's enough to put the argument in one line ( < 80 chars)

File: safe/gui/widgets/dock.py [hunk 2]
```diff
@@ -1319,6 +1322,77 @@ def prepare_analysis(self):
...
+
+        .. note:: This method works in QGIS 2.4 and better only. In
+            earlier versions it will just add the layer to the top of the
+            layer stack.
+
+        .. seealso:: issue #2322
+
+        :param existing_layer: The layer which the new layer
+            should be added above.
+        :type new_layer: QgsMapLayer
```
Comment (ismailsunni) [inline safe/gui/widgets/dock.py]: should it be `existing_layer`?

**Comment by ismailsunni** [comment]
LGTM with some minor comments

**Pull request merged**